src/nifreeze/data/dmri.py

```diff
@@ -369,10 +369,10 @@ def load(
 
 
 def find_shelling_scheme(
-    bvals,
-    num_bins=DEFAULT_NUM_BINS,
-    multishell_nonempty_bin_count_thr=DEFAULT_MULTISHELL_BIN_COUNT_THR,
-    bval_cap=DEFAULT_HIGHB_THRESHOLD,
+    bvals: np.ndarray,
+    num_bins: int = DEFAULT_NUM_BINS,
+    multishell_nonempty_bin_count_thr: int = DEFAULT_MULTISHELL_BIN_COUNT_THR,
+    bval_cap: int = DEFAULT_HIGHB_THRESHOLD,
 ):
```

Comment:
It would be good to specify return types, too. Looks (without checking) like:

```suggestion
) -> tuple[str, list[npt.NDArray[np.floating]], list[np.floating]]:
```

You'll need to `import numpy.typing as npt`.